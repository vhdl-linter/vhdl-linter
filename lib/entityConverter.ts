import { Position } from 'vscode-languageserver';
import { findObjectFromPosition } from './languageFeatures/findObjects';
import { OArchitecture, OEntity, OGeneric, OPort } from './parser/objects';
import { ISettings } from './settingsGenerated';
import { VhdlLinter } from './vhdlLinter';


function longestInArray(array: OPort[] | OGeneric[]) {
    let longest = 0;
    for (const object of array) {
        if (object.lexerToken.text.length > longest) {
            longest = object.lexerToken.text.length;
        }
    }
    return longest;
}
export function instanceTemplate(entity: OEntity, settings: ISettings) {
    let text = `${settings.style.instantiationLabelPrefix}${entity.lexerToken.text}${settings.style.instantiationLabelSuffix} : entity work.${entity.lexerToken.text}`;
    const indentString = '  ';
    if (entity.generics.length > 0) {
        text += `\ngeneric map (\n`;
        const longest = longestInArray(entity.generics);
        for (const generic of entity.generics) {
            const name = generic.lexerToken.text.padEnd(longest, ' ');
            text += `${indentString}${name} => ${settings.style.genericPrefix}${generic.lexerToken.text}${settings.style.genericSuffix},\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n)`;
    }

    if (entity.ports.length > 0) {
        text += `\nport map (\n`;
        const longest = longestInArray(entity.ports);
        for (const port of entity.ports) {
            const name = port.lexerToken.text.padEnd(longest, ' ');
            text += `${indentString}${name} => ${settings.style.signalPrefix}${port.lexerToken.text}${settings.style.signalSuffix},\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n)`;
    }

    text += `;\n`;
    return text;
}
export function componentTemplate(entity: OEntity) {
    let text = `component ${entity.lexerToken.text} is`;
    const indentString = '  ';
    if (entity.generics.length > 0) {
        text += `\ngeneric (\n`;
        for (const generic of entity.generics) {
            text += `${indentString}${generic.range.getText()};\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n);`;
    }

    if (entity.ports.length > 0) {
        text += `\nport (\n`;
        for (const port of entity.ports) {
            text += `${indentString}${port.range.getText()};\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n);`;
    }

    text += `\nend component;\n`;
    return text;
}

export function sysVerilogTemplate(entity: OEntity, settings: ISettings) {
    let text = entity.lexerToken.text;
    const indentString = '  ';
    if (entity.generics.length > 0) {
        text += ` #(\n`;
        const longest = longestInArray(entity.generics);
        for (const generic of entity.generics) {
            const name = generic.lexerToken.text.padEnd(longest, ' ');
            text += `${indentString}.${name}(${settings.style.genericPrefix}${generic.lexerToken.text}${settings.style.genericSuffix}),\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n)`;
    }

    if (entity.ports.length > 0) {
        text += ` ${settings.style.instantiationLabelPrefix}${entity.lexerToken.text}${settings.style.instantiationLabelSuffix} (\n`;
        const longest = longestInArray(entity.ports);
        for (const port of entity.ports) {
            const name = port.lexerToken.text.padEnd(longest, ' ');
            text += `${indentString}.${name}(${settings.style.signalPrefix}${port.lexerToken.text}${settings.style.signalSuffix}),\n`;
        }
        // Strip the final comma
        text = text.slice(0, -2);
        text += `\n)`;
    }

    text += `;\n`;
    return text;
}
export function signalsTemplate(entity: OEntity, settings: ISettings) {
    let text = '';
    if (entity.ports.length > 0) {
        const longest = longestInArray(entity.ports);
        for (const port of entity.ports) {
            const name = port.lexerToken.text.padEnd(longest, ' ');
            let typeText: string | undefined;
            if (port.subtypeIndication.typeNames.length > 0) {
                typeText = port.subtypeIndication.typeNames[0]!.range.copyWithNewEnd(port.range.end).getText();
            }
            text += `signal ${settings.style.signalPrefix}${name}${settings.style.signalSuffix} : ${typeText ?? 'unknown_type'};\n`;
        }
    }
    return text;
}
export type converterTypes = 'instance' | 'signals' | 'sysverilog' | 'component';
export function entityConverter(vhdlLinter: VhdlLinter, type: converterTypes, settings: ISettings, position?: Position) {
    let entity;
    if (position === undefined) {
        entity = vhdlLinter.file.entities[0];
    } else {
        const rootElement = findObjectFromPosition(vhdlLinter, position)[0]?.getRootElement();
        if (rootElement instanceof OEntity) {
            entity = rootElement;
        } else if (rootElement instanceof(OArchitecture) && rootElement.correspondingEntity) {
            entity = rootElement.correspondingEntity;
        } else {
            // Fallback to first entity
            entity = vhdlLinter.file.entities[0];
        }
    }
    if (!entity) {
        return;
    }
    if (type === 'instance') {
        return instanceTemplate(entity, settings);
    } else if (type === 'signals') {
        return signalsTemplate(entity, settings);
    } else if (type === 'sysverilog') {
        return sysVerilogTemplate(entity, settings);
    } else if (type === 'component') {
        return componentTemplate(entity);
    }
}