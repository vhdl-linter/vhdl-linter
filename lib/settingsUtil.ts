import { DeepPartial } from "utility-types";
import { ISettings } from "./settingsGenerated";
import { existsSync, readFileSync } from "fs";
import { join } from "path";


// these are capabilities from the language server and mainly used by the language server
// However, the project parser might use the `configuration` to determine whether to get settings from the language server.
// We define them here and not in the languageServer to not start one from the projectParser
export const currentCapabilities = {
  configuration: false,
  workspaceFolder: false
};

export function normalizeSettings(settings: ISettings) {
  const newSettings = JSON.parse(JSON.stringify(settings)) as ISettings;
  for (const [key, value] of Object.entries(newSettings.style)) {
    if (typeof (newSettings.style as Record<string, unknown>)[key] === 'string') {
      (newSettings.style as Record<string, unknown>)[key] = (value as string).trim();
    }
  }
  return newSettings;
}

export function overwriteSettings(settings: ISettings, overwrite: DeepPartial<ISettings>) {
  const newSettings = JSON.parse(JSON.stringify(settings)) as ISettings;
  recursiveObjectAssign(newSettings, overwrite);
  return normalizeSettings(newSettings);
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
function recursiveObjectAssign<T extends Record<string, any>>(target: T, source: DeepPartial<T>) {
  Object.keys(source).forEach(key => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const s_val = source[key];
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const t_val = target[key];
    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/strict-boolean-expressions
    (target as any)[key] = t_val && s_val && typeof t_val === 'object' && typeof s_val === 'object'
      ? recursiveObjectAssign(t_val, s_val)
      : s_val;
  });
  return target;
}
type primitive = string | boolean | number;
interface Schema {
  [key: string]: Schema | primitive[] | primitive;
}
// for initial transpile before doing ts-node generateSettings, the file does not exist...
const schemaFile = join(__dirname, '../settings.schema.json');
const file = existsSync(schemaFile) ? JSON.parse(readFileSync(join(__dirname, '../settings.schema.json'), { encoding: 'utf-8' })) as Schema : {};
function objectWalk(object: Schema) {
  const result: Schema = {};
  for (const [key, value] of Object.entries(object)) {
    if (key !== 'deprecationMessage') {
      if (typeof value === 'object' && !Array.isArray(value)) {
        result[key] = objectWalk(value);
      } else {
        result[key] = value;
      }
    }
  }
  return result;
}
export const settingsSchema = objectWalk(file);