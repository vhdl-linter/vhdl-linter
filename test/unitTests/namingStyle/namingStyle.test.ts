import { expect, test } from '@jest/globals';
import { join } from 'path';
import { pathToFileURL } from 'url';
import { ProjectParser } from '../../../lib/projectParser';
import { VhdlLinter } from '../../../lib/vhdlLinter';
import { readFileSyncNorm } from "../../readFileSyncNorm";
import { overwriteSettings } from '../../../lib/settingsUtil';
test.each([
  'signal',
  'variable',
  'constant',
  'generic',
  'out',
  'in',
  'inout',
  'parameterOut',
  'parameterIn',
  'parameterInout',
  'instantiationLabel',
  'unused'
].flatMap(file => [[file, 'pre_', ''], [file, '', '_suf']]))('testing naming style for %s with prefix "%s" and suffix "%s"', async (file: string, prefix: string, suffix: string) => {
  const overwrite: Record<string, string> = {};
  overwrite[`${file}Prefix`] = prefix;
  overwrite[`${file}Suffix`] = suffix;
  const path = join(__dirname, `${file}.vhd`);
  const projectParser = await ProjectParser.create([pathToFileURL(__dirname)]);
  const settings = overwriteSettings(await projectParser.getDocumentSettings(pathToFileURL(path)) , {
    style: overwrite
  });
  const linter = new VhdlLinter(pathToFileURL(path), readFileSyncNorm(path, { encoding: 'utf8' }),
    projectParser, settings);
  await linter.checkAll();

  expect(linter.messages).toMatchSnapshot();
  const codes = linter.messages.filter(message => file ==='unused' ? message.message.includes('(unused)') : message.message.includes('(naming-style)')).map(message => String(message.code ?? '').split(String(';'))).flat();
  const solutions = (await Promise.all(codes.map(async code => await linter.diagnosticCodeActionResolveRegistry[parseInt(code)]?.(linter.uri.toString())))).flat()
    .filter(solution => solution?.title.includes('Replace with'));
  expect(solutions.map(solution => ({
    ...solution,
    edit: solution?.edit?.changes?.[pathToFileURL(path).toString()],

  }))).toMatchSnapshot();
  await projectParser.stop();
});