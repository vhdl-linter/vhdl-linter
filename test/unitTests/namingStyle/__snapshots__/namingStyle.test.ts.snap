// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing naming style for constant with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for constant (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 16,
        "line": 1,
      },
      "start": {
        "character": 11,
        "line": 1,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for constant with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 16,
            "line": 1,
          },
          "start": {
            "character": 11,
            "line": 1,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for constant with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for constant (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 16,
        "line": 1,
      },
      "start": {
        "character": 11,
        "line": 1,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for constant with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 16,
            "line": 1,
          },
          "start": {
            "character": 11,
            "line": 1,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for generic with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for generic (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for generic with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 30,
            "line": 9,
          },
          "start": {
            "character": 25,
            "line": 9,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for generic with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for generic (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for generic with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 30,
            "line": 9,
          },
          "start": {
            "character": 25,
            "line": 9,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for instantiationLabel with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for instantiation label (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 7,
        "line": 4,
      },
      "start": {
        "character": 2,
        "line": 4,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for instantiationLabel with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": undefined,
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for instantiationLabel with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for instantiation label (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 7,
        "line": 4,
      },
      "start": {
        "character": 2,
        "line": 4,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for instantiationLabel with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": undefined,
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for parameterIn with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for input parameter (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterIn with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for parameterIn with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for input parameter (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterIn with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for parameterInout with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for inout parameter (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterInout with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for parameterInout with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for inout parameter (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterInout with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for parameterOut with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for output parameter (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterOut with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for parameterOut with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for output parameter (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for parameterOut with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for portIn with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for input port (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portIn with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for portIn with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for input port (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portIn with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for portInout with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for inout port (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portInout with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for portInout with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for inout port (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portInout with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for portOut with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for output port (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portOut with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for portOut with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for output port (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 9,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for portOut with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 9,
            "line": 2,
          },
          "start": {
            "character": 4,
            "line": 2,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for signal with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for signal (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 14,
        "line": 3,
      },
      "start": {
        "character": 9,
        "line": 3,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for signal with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 14,
            "line": 3,
          },
          "start": {
            "character": 9,
            "line": 3,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for signal with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for signal (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 14,
        "line": 3,
      },
      "start": {
        "character": 9,
        "line": 3,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for signal with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 14,
            "line": 3,
          },
          "start": {
            "character": 9,
            "line": 3,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for unused with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "Not writing signal 'apple' (unused)",
    "range": {
      "end": {
        "character": 14,
        "line": 3,
      },
      "start": {
        "character": 9,
        "line": 3,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for unused with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 14,
            "line": 3,
          },
          "start": {
            "character": 9,
            "line": 3,
          },
        },
      },
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 30,
            "line": 7,
          },
          "start": {
            "character": 25,
            "line": 7,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for unused with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "Not writing signal 'apple' (unused)",
    "range": {
      "end": {
        "character": 14,
        "line": 3,
      },
      "start": {
        "character": 9,
        "line": 3,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for unused with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 14,
            "line": 3,
          },
          "start": {
            "character": 9,
            "line": 3,
          },
        },
      },
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 30,
            "line": 7,
          },
          "start": {
            "character": 25,
            "line": 7,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;

exports[`testing naming style for variable with prefix "" and suffix "_suf" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for variable (name_suf) (naming-style)",
    "range": {
      "end": {
        "character": 16,
        "line": 1,
      },
      "start": {
        "character": 11,
        "line": 1,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for variable with prefix "" and suffix "_suf" 2`] = `
[
  {
    "edit": [
      {
        "newText": "apple_suf",
        "range": {
          "end": {
            "character": 16,
            "line": 1,
          },
          "start": {
            "character": 11,
            "line": 1,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'apple_suf'",
  },
]
`;

exports[`testing naming style for variable with prefix "pre_" and suffix "" 1`] = `
[
  {
    "code": "0;1",
    "message": "apple does not match the style settings for variable (pre_name) (naming-style)",
    "range": {
      "end": {
        "character": 16,
        "line": 1,
      },
      "start": {
        "character": 11,
        "line": 1,
      },
    },
    "severity": 2,
  },
]
`;

exports[`testing naming style for variable with prefix "pre_" and suffix "" 2`] = `
[
  {
    "edit": [
      {
        "newText": "pre_apple",
        "range": {
          "end": {
            "character": 16,
            "line": 1,
          },
          "start": {
            "character": 11,
            "line": 1,
          },
        },
      },
    ],
    "kind": "quickfix",
    "title": "Replace with 'pre_apple'",
  },
]
`;
